resources/assets/js/components/JobTasks/JobTasks.tsx

```diff
@@ -1,222 +1,387 @@
+            )}
+            <Form id="job-tasks">
+              <FieldArray
+                name="tasks"
+                render={(arrayHelpers): React.ReactElement => (
+                  <>
+                    <div data-c-grid="gutter">
+                      {values.tasks &&
+                        values.tasks.length > 0 &&
+                        values.tasks.map(
+                          (task, index): React.ReactElement => (
+                            <>
+                              {validCount! === index && (
+                                <div
+                                  className="job-builder-task-warning"
+                                  data-c-grid-item="base(1of1)"
+                                >
+                                  <div
+                                    data-c-alert="error"
+                                    data-c-radius="rounded"
+                                    role="alert"
+                                    data-c-margin="bottom(normal)"
+                                  >
+                                    <div data-c-padding="half">
+                                      <span
+                                        data-c-margin="bottom(quarter)"
+                                        data-c-heading="h5"
+                                        data-c-font-weight="bold"
+                                      >
+                                        <i
+                                          aria-hidden="true"
+                                          className="fas fa-exclamation-circle"
+                                        />
+                                        <FormattedMessage
+                                          id="jobTasks.taskCount.error.title"
+                                          description="Error message displayed when too many tasks are on screen."
+                                          defaultMessage="Just a heads up!"
+                                        />
+                                      </span>
+                                      <p>
+                                        <FormattedMessage
+                                          id="jobTasks.taskCount.error.body"
+                                          description="Error message displayed when too many tasks are on screen."
+                                          defaultMessage="You have exceeded the maximum number of key tasks allowed, but that’s okay. You can continue to add key tasks as you brainstorm here, but you will be asked to trim your list to 6 key tasks or fewer to proceed."
+                                        />
+                                      </p>
+                                    </div>
+                                  </div>
+                                </div>
+                              )}
+                              <div
+                                key={`listItem${index}`}
```

Comment:
Its dangerous to use index as the key, _particularly_ when the order will change. Is it possible to give new tasks a temporary unique id somehow? 
https://reactjs.org/docs/lists-and-keys.html#keys

Reply:
I'll see what I can do. We'll have to figure out some sort of temp ID standard, unless we're just worried about having it as the key in isolated cases like this.

Reply:
As we discussed I added the [nanoid](https://github.com/ai/nanoid) library for generating temp IDs. We'll keep it to 10 characters for now as that seems to be more than fine.

I'm still seeing console warnings for missing keys inside of a loop, so I'm not sure if we'll still see unnecessary rerenders inside the tree. I think it has to do with how I'm using fragments inside the FieldArray but no combination of adding keys to any element inside of those loops seem to resolve it.